Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/MainActivity.java

```diff
@@ -113,8 +113,12 @@ private void analyze() {
     totalFrames =
         Integer.valueOf(mediaMetadataRetriever.extractMetadata(METADATA_KEY_VIDEO_FRAME_COUNT));
     // TODO: Add on demand frame read opposed to read all frames.
-    frameList = mediaMetadataRetriever.getFramesAtIndex(0, Math.max(0, totalFrames - SAFE_FRAMES));
```

Comment:
Any details on why this crashed?  did you get an excpetion message?

I would of assumed it was because it was too large for memory, but in that case the new code should crash too, so I'm not sure what the problem is.

Reply:
@mwarton Added a log to catch the exception.
Added details to the issue discussion.